State.lua

```diff
@@ -1066,6 +1066,27 @@ local function removeBuff( aura )
 end
 state.removeBuff = removeBuff
 
+-- Extend buff function, defaults to 1 second if no value supplied, breaks out if supplied a 0
+local function extendBuff( aura, duration)
+
+    if duration == 0 then
+        Error( "Attempted to extend a buff '%s'.\n\n%s by 0 seconds", aura or "nil", debugstack() )
+        return
+    end
```

Comment:
Why is this check needed? Your function already accepts positive and negative times. Accepting zero seems okay to me, too?

Reply:
Why waste calculations looking up a buff and trying to add 0 to it? positive values extend a buff, negative values reduce it (like warlock ritual), 0 wastes CPU time.

Reply:
The question is why it's an error. You can just short-circuit and return immediately if it's passed zero as an argument.

Reply:
ah, gotcha.